exporters/deploytime/app.py

```diff
@@ -37,39 +38,41 @@ def collect(self) -> Iterable[GaugeMetricFamily]:
             "Deployment timestamp",
             labels=["namespace", "app", "image_sha"],
         )
-        deployment_active_metric = GaugeMetricFamily(
-            "deployment_active",
-            "Active deployments in cluster",
-            labels=["namespace", "app", "image_sha"],
-        )
+
+        number_of_dropped = 0
 
         for m in metrics:
+            if not is_out_of_date(str(m.deploy_time_timestamp)):
+                logging.info(
+                    "Collected deploy_timestamp{namespace=%s, app=%s, image=%s} %s (%s)",
+                    m.namespace,
+                    m.name,
+                    m.image_sha,
+                    m.deploy_time_timestamp,
+                    m.deploy_time,
+                )
+                deploy_timestamp_metric.add_metric(
+                    [m.namespace, m.name, m.image_sha],
+                    m.deploy_time_timestamp,
+                    timestamp=m.deploy_time_timestamp,
+                )
+            else:
+                number_of_dropped += 1
+                logging.debug(
+                    "Deployment too old to be collected: deploy_timestamp{namespace=%s, app=%s, image=%s} %s (%s)",
+                    m.namespace,
+                    m.name,
+                    m.image_sha,
+                    m.deploy_time_timestamp,
+                    m.deploy_time_timestamp,
+                )
+        if number_of_dropped:
             logging.info(
```

Comment:
I do not about this log, can it be too often called?

Reply:
moved to debug. you are right.